android/src/main/java/com/reactnativecommunity/cameraroll/Utils.java

```diff
@@ -0,0 +1,38 @@
+package com.reactnativecommunity.cameraroll;
+
+import android.content.ContentResolver;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.MediaStore;
+
+import javax.annotation.Nullable;
+
+public class Utils {
+
+  // From https://stackoverflow.com/a/64359655/1377145
+  @Nullable
+  public static String getFilePathFromContentUri(Context context, Uri contentUri) {
```

Comment:
Replacing this with [this other answer](https://stackoverflow.com/a/11603899) on the same SO post fixes the null document_id issue

Reply:
can you make sure it works on android9, 10 and 11?

Reply:
I'll see if I have time to do so. right now I'm using a custom native module just like yours, which checks the api target level and defaults to CameraRoll if it is < 11